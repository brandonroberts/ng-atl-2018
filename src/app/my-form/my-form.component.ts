import { Component, OnInit } from '@angular/core';
import { FormControl, Validators, FormGroup } from '@angular/forms';

@Component({
  selector: 'app-my-component',
  template: `
    <form [formGroup]="form">
      <input placeholder="Email" formControlName="email">
      <span *ngIf="form.hasError('required', ['email'])">
        Email is required
      </span>
    </form>
  `
})
export class MyFormComponent implements OnInit {
  form = new FormGroup({
    email: new FormControl('', [Validators.required])
  });

  constructor() { }

  ngOnInit() {
  }

}
