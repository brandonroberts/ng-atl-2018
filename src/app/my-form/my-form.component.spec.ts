import { ComponentFixture, TestBed } from '@angular/core/testing';

import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ReactiveFormsModule } from '@angular/forms';
import { MyFormComponent } from './my-form.component';
import { createComponentFixture } from '../test-helpers';

describe('MyFormComponent', () => {
  const fixture = createComponentFixture({
    component: MyFormComponent,
    imports: [
      ReactiveFormsModule
    ]
  });

  beforeEach(() => {
    fixture.compile();
  });

  it('should compile', () => {
    expect(fixture).toMatchSnapshot();
  });
});
