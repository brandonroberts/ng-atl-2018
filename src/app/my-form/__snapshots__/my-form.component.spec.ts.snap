// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFormComponent should compile 1`] = `
<app-form-component
  form={[Function FormGroup]}
>
  
      <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    
        
    <input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="email"
      ng-reflect-name="email"
      placeholder="Email"
    />
    
        
    
    <span>
      
          Email Address is required
        
    </span>
    
      
  </form>
    
</app-form-component>
`;
